geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -94,6 +101,16 @@ public RedisSet(int size) {
     return diff;
   }
 
+  private static int setOpStoreResult(RegionProvider regionProvider, RedisKey destinationKey,
+      Set<byte[]> diff) {
+    if (diff.isEmpty()) {
+      regionProvider.getDataRegion().remove(destinationKey);
+    } else {
+      regionProvider.getLocalDataRegion().put(destinationKey, new RedisSet(diff));
```

Comment:
for the "region.put" case you should do a check to see if destinationKey exists. If it doesn't do the put. But otherwise modify the existing RedisSet in place and call storeChanges with a DeltaInfo. You will want to merge in the latest develop to get the fix for GEODE-9772. The new DeltaType could be SET_BYTE_ARRAYS and it would just be a List of byte[] like like ADD_BYTE_ARRAYS. The difference is that when this DeltaInfo is applied you will want to first call something on the RedisSet that clears it and calls persistNoDelta and then just calls applyAddByteArrayDelta repeatedly. You will also need to add a method on the near side (not the delta apply side) that does this (given a Set<byte[]> does persistNoDelta() and "members = new MemberSet(diff)". Even better would be to make sure calculateDIff always returns a MemberSet instead of a Set and then this could just be "member = diff". I think the only case when it is not a MemberSet is when it is empty in which case you do a region.remove so in this code you could just cast diff to MemberSet safely.
I can help with this delta work if you have any questions.

Reply:
Thanks! I'll let you know if I need help. 

I was following zunionstore and zinterstore in RedisSortedSet that was using this method of storing the members of the key. Should a JIRA ticket be create to change how those members are stored? I feel like another store method should exist somewhere, maybe AbstractRedisData, to handle the case where it overwrites the destination key.